pkg/platform/kube/test/platform_test.go

```diff
@@ -1423,6 +1423,49 @@ func (suite *DeployFunctionTestSuite) TestDeployFunctionWithSidecarSanity() {
 	})
 }
 
+func (suite *DeployFunctionTestSuite) TestDeployFunctionWithInitContainers() {
```

Comment:
Please also add bad flow test, maybe by verifying the logs/message in the function status available in the deploy result

Reply:
@TomerShor in that case we won't deploy a function(main pod) at all, and we won't have a right message in the crd since controller is not running. Also, in normal flow we could not predict which issue we will catch - still running init container (after termination+restart) or its termination. I've tested some options manually and they are fine.